include/deal.II/base/lazy.h

```diff
@@ -314,72 +287,88 @@ inline Lazy<T>::Lazy()
 {}
 
 
+
 template <typename T>
 DEAL_II_CXX20_REQUIRES((std::is_move_constructible_v<T> &&
                         std::is_move_assignable_v<T>))
 inline Lazy<T>::Lazy(const Lazy &other)
-  : object(other.object)
 {
-  object_is_initialized.store(other.object_is_initialized.load());
+  // If the other object has a value stored, then get it and set our
+  // own object to a copy of it:
+  if (other.has_value())
+    {
+      object_is_initialized.store(true);
+      task_result =
+        Threads::new_task([&other]() -> T { return other.value(); });
+      task_result.join();
+    }
```

Comment:
This is not optimal: Running a separate task that only copies a value. There is currently no other way to get a value into a `TaskResult` object. We could add a function `TaskResult::try_emplace_value()` that sets a value if none has been set so far. Thoughts?

Reply:
should this be try_emplace_task() instead of operator=?

Reply:
We're in the constructor of `Lazy` here, so we do not have to worry that another thread is accessing the object. So `operator=` is ok.

Reply:
I will fix this in a follow-up PR that I'm currently testing.

Reply:
See #17592.

Reply:
do you want to update this pr now that #17592 is merged?

Reply:
do you want to update this pr now that #17592 is merged?

Reply:
Let me leave this for a follow-up that I've already got queued on a machine somewhere (but not the one I'm on right now).